### ISSUE TITLE
input_ids is None when evaluating

### ISSUE THREAD

**Issue by qshi95** [huggingface/transformers#14228]
Information
=====
When evaluating model checkpoints, I find that input_ids passed to the model is None, but encoder_outputs are available instead. I'd like to obtain the value of input_ids, what should I do? Thanks.

To reproduce
=====
run the following command:

python run_summarization.py \
    --model_name_or_path facebook/bart-large \
    --do_eval \
    --dataset_name cnn_dailymail \
    --dataset_config "3.0.0" \
    --source_prefix "summarize: " \
    --output_dir /tmp/tst-summarization \
    --per_device_train_batch_size=4 \
    --per_device_eval_batch_size=4 \
    --overwrite_output_dir \
    --predict_with_generate

and print(input_ids) in the forward functions of the model.

**Comment by qshi95** [comment]
I find that evaluating will run a bart encoder in advance, I add a self.input_ids = input_ids in the encoder, and then can obtain it easily.

**Comment by JaeeonPark** [comment]
Hi, I am also trying to get input_ids in evaluation mode. I am a bit confused about your solution. 
How can we plug input_ids into the encoder?
`class BartDecoder(BartPretrainedModel):
    def __init__(self, input_ids, config: BartConfig, embed_tokens: Optional[nn.Embedding] = None):
        super().__init__(config)
        self.input_ids = input_ids
`
Is it right what you mentioned? 
Thanks in advance.
